tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -14,9 +15,12 @@
 
 namespace Datadog.Trace.Agent.MessagePack
 {
-    internal class SpanMessagePackFormatter : IMessagePackFormatter<Span>
+    internal class SpanMessagePackFormatter : IMessagePackFormatter<TraceChunkModel>,
+                                              IMessagePackFormatter<SpanModel>,
+                                              IMessagePackFormatter<Span>, // deprecated, use IMessagePackFormatter<SpanModel>
+                                              IMessagePackFormatter<ArraySegment<Span>> // deprecated, use IMessagePackFormatter<TraceChunkModel>
```

Comment:
Are these actually used anywhere after this change?

Reply:
Some tests serialize spans directly and then deserialize them to check their contents.

TODO: update those tests and remove these old implementations before merging this PR, since we want the tests to use the same serialization code.